language_generators/scala-defref-extractor/src/main/scala/io/bazeltools/buildgen/shared/DriverApplication.scala

```diff
@@ -92,6 +92,22 @@ abstract class DriverApplication extends IOApp {
     if (parallel) parallelExtractDataBlocks(workingDirectory, paths)
     else sequentialExtractDataBlocks(workingDirectory, paths)
 
+  private[this] def isValidTld(s: String): Boolean =
+    s.matches("^[a-z]+$")
+
+  private[this] def setSpecialTlds(specialTldsEnv: String): IO[Unit] = {
+    val names = specialTldsEnv.split(',').toList
+    if (names.forall(isValidTld)) {
+      IO(Entity.setSpecialTlds(names))
+    } else {
+      val invalid = names.filter(s => !isValidTld(s))
+      IO.raiseError(new Exception(s"invalid TLDs: $invalid"))
```

Comment:
Test failures may be showing that we raise an error in the case of `specialTldsEnv` being empty?